### PR TITLE
GraceDB code

### PR DESCRIPTION
**Pull request by EthanMarx**
Add gracedb upload code. @alecgunny

### CODE REVIEW

File: projects/sandbox/deploy/deploy/main.py [hunk 1]
```diff
@@ -1,25 +1,59 @@
...
-    return
+def connect_to_gracedb(server: Gdb_server):
+    if server in ["playground", "test"]:
+        server = f"https://gracedb-{server}.ligo.org/api/"
+    elif server == "production":
+        server = "https://gracedb.ligo.org/api/"
+    else:
+        raise ValueError(f"Unknown server {server}")
+
+    gdb = GraceDb(service_url=server)
```
Comment (wbenoit26) [inline projects/sandbox/deploy/deploy/main.py:32]: Do we need to provide credentials here?

File: projects/sandbox/deploy/deploy/main.py [hunk 2]
```diff
@@ -1,25 +1,59 @@
...
+    filename = "event.json"
+    filecontents = str(asdict(trigger))
+    # alternatively we can write a file to disk,
+    # pass that path to the filename argument,
+    # and set filecontents=None
+    response = gdb.createEvent(
+        group="CBC",
+        pipeline="BBHNet",
+        filename=filename,
+        search="BBHNet",
```
Comment (wbenoit26) [inline projects/sandbox/deploy/deploy/main.py:53]: This should just be "BBH"

Comment (EthanMarx) [reply projects/sandbox/deploy/deploy/main.py:53]: I don't think there is just a "BBH" group. See https://gracedb-test.ligo.org/documentation/models.html?highlight=group

File: projects/sandbox/deploy/deploy/main.py (hunk 1 above)
Comment (EthanMarx) [reply projects/sandbox/deploy/deploy/main.py:32]: I think we can provide them explicitly, but it should also pick them up automatically via environment variables (see https://ligo-gracedb.readthedocs.io/en/latest/api.html). However we do need GraceDB folks to authorize specific users (i.e Will Ethan Rafia Alec etc.) to upload a "BBHnet" (or whatever we end up calling it) event. 

Down the line, we'll likely make a ligo "BBHnet" account from which we will run production analysis and submit triggers.

File: projects/sandbox/deploy/deploy/main.py (hunk 2 above)
Comment (wbenoit26) [reply projects/sandbox/deploy/deploy/main.py:53]: Right, the group is "CBC", but the search should be "BBH"

Comment (EthanMarx) [reply projects/sandbox/deploy/deploy/main.py:53]: Ah yeah I see.

Comment (alecgunny) [reply projects/sandbox/deploy/deploy/main.py:53]: Good catch. I'll update this on my PR and then close this one, which has integrated the changes here

**Pull request closed without merging**